e2e/container/run_test.go

```diff
@@ -149,3 +150,40 @@ func TestRunWithCgroupNamespace(t *testing.T) {
 		"/bin/grep", "-q", "':memory:/$'", "/proc/1/cgroup")
 	result.Assert(t, icmd.Success)
 }
+
+func TestMountSubvolume(t *testing.T) {
+	t.SkipNow() // TODO: Enable when testing against 1.44+
+	volName := "test-volume-" + t.Name()
+	icmd.RunCommand("docker", "volume", "create", volName).Assert(t, icmd.Success)
+
+	t.Cleanup(func() {
+		icmd.RunCommand("docker", "volume", "remove", "-f", volName).Assert(t, icmd.Success)
+	})
+
+	defaultMountOpts := []string{
+		"type=volume",
+		"src=" + volName,
+		"dst=/volume",
+	}
+
+	// Populate the volume with test data.
+	icmd.RunCommand("docker", "run", "--mount", strings.Join(defaultMountOpts, ","), fixtures.AlpineImage, "sh", "-c",
+		"echo foo > /volume/bar.txt && "+
+			"mkdir /volume/subdir && echo world > /volume/subdir/hello.txt;",
+	).Assert(t, icmd.Success)
+
+	runMount := func(cmd string, mountOpts ...string) *icmd.Result {
+		mountArg := strings.Join(append(defaultMountOpts, mountOpts...), ",")
+		return icmd.RunCommand("docker", "run", "--mount", mountArg, fixtures.AlpineImage, cmd, "/volume")
+	}
+
+	t.Run("subpath not exists", func(t *testing.T) {
+		runMount("ls", "volume-subpath=some-path/that/doesnt-exist").Assert(t, icmd.Expected{Err: "volume's path is not accessible", ExitCode: 125})
+	})
+	t.Run("subdirectory mount", func(t *testing.T) {
+		runMount("ls", "volume-subpath=subdir").Assert(t, icmd.Expected{Out: "hello.txt"})
```

Comment:
I assume these subpaths are always relative to the volume's root path, is that correct? Can absolute paths ever be used here? (if so I'd add some tests)

Reply:
Yes they must be relative. Absolute paths are not allowed as we only want to mount things that are inside the volume.

Reply:
Perhaps a test-case would be ok indeed; make sure that we get an error when trying to do so (we could add a test-table with good / bad variants).


Actually; looking at the test-cases; I wonder if a test-table would be a good idea overall (`subpath`, `expectedOut`, `expectedErr`, or something along those lines; it could make it easier to add additional test-cases).

Reply:
Done